src/Admin/AbstractAdmin.php

```diff
@@ -1871,6 +1877,20 @@ public function getChild($code)
     public function setParent(AdminInterface $parent)
     {
         $this->parent = $parent;
+
+        // NEXT_MAJOR: remove $args and add $field parameter to this function on next Major
+
+        $args = \func_get_args();
+
+        if (isset($args[1])) {
+            $this->parentAssociationMapping[$parent->getCode()] = $args[1];
+        } else {
+            @trigger_error(
+                'Calling "setParent" without second argument is deprecated since'
+                .' sonata-project/admin-bundle 3.x and will not be allowed in 4.0.',
+                E_USER_DEPRECATED
+            );
```

Comment:
```suggestion
            @trigger_error(sprintf(
                'Not passing argument 2 to "%s()" is deprecated since'
                .' sonata-project/admin-bundle 3.x and will not be allowed in 4.0.',
                __METHOD__
            ), E_USER_DEPRECATED);
```